### ISSUE TITLE
[BUG] Created video file still locked after context has been closed

### ISSUE THREAD

**Issue by LanderBeeuwsaert** [microsoft/playwright#5231]
**Context:**
- Playwright Version: 1.8
- Operating System: Windows
- Node.js version: v14.15.1
- Browser: WebKit

We get an error when running the following piece of code if the browser in which this context is launched is webkit.
We don't get this error when running with chromium.
In essence what we do is that we rename the video file that was created.
However, this doesn't work for webkit as the video file is still locked.
This doesn't happen for all test suites, but it consistently happens for this one, most probably because this one is +- 4 minutes long, so video file creation probably takes a long time.

```js
    await this.breakdown();
    if (this.createVideo) {
      const videoFilePath = await this.page.video().path();
      const newVideoPath = videoFilePath.replace(/\\(\w)*.webm/g, '\\balance_group_testsuite.webm');
      fs.renameSync(videoFilePath, newVideoPath);
    }

  async breakdown() {
    if (this.createVideo) Log.log('important', 'Video (balance_group_testsuite) -> Create start');
    await this.world.destroy();
    await this.page.close();
    await this.context.close();
    if (this.createVideo) Log.log('important', 'Video (balance_group_testsuite) -> Create end');
  }
```

```
Error: EBUSY: resource busy or locked, rename 'C:\Lander\WS\acro-companion\e2e-testing\playwright\videos\c3a0f619266526b7282a998b4b7c959a.webm' -> 'C:\Lander\WS
\acro-companion\e2e-testing\playwright\videos\balance_group_testsuite.webm'
    at Object.renameSync (fs.js:772:3)
    at BalanceGroupTestsuite.<anonymous> (C:\Lander\WS\acro-companion\e2e-testing\playwright\tests\balance-group-testsuite.spec.ts:68:12)
    at step (C:\Lander\WS\acro-companion\e2e-testing\playwright\tests\balance-group-testsuite.spec.ts:42:23)
    at Object.next (C:\Lander\WS\acro-companion\e2e-testing\playwright\tests\balance-group-testsuite.spec.ts:23:53)
    at fulfilled (C:\Lander\WS\acro-companion\e2e-testing\playwright\tests\balance-group-testsuite.spec.ts:14:58) {
  errno: -4082,
  syscall: 'rename',
  code: 'EBUSY',
  path: 'C:\\Lander\\WS\\acro-companion\\e2e-testing\\playwright\\videos\\c3a0f619266526b7282a998b4b7c959a.webm',
  dest: 'C:\\Lander\\WS\\acro-companion\\e2e-testing\\playwright\\videos\\balance_group_testsuite.webm'
}
```

**Comment by yury-s** [comment]
@LanderBeeuwsaert can you run the test with `DEBUG=pw:protocol` and send us the log?

**Comment by LanderBeeuwsaert** [comment]
@yury-s Jep, I'll try to do it this weekend.

**Comment by LanderBeeuwsaert** [comment]
@yury-s sooner than expected:
[playwright log.txt](https://github.com/microsoft/playwright/files/5909224/playwright.log.txt)

**Comment by LanderBeeuwsaert** [comment]
it's probably only the last few seconds, but the relevant part is there.
Not sure it matters, I'm running at that time 3 webkit contexts from the same browser instance in parallel.

**Comment by yury-s** [comment]
Thanks for sharing the trace! Is it possible that that the page in `await this.page.video().path();` doesn't belong to the context being closed in `breakdown()` ? Looking at the trace I see `Playwright.screencastFinished` event (which means that the video file should be closed already) and then one of the contexts is destroyed. Since the trace is incomplete it's impossible to tell video from which page has finished. Have you seen this problem with a single context or only when running several of them in parallel?

**Comment by LanderBeeuwsaert** [comment]
I've not tried with a single context. However we're running this same code already months with chromium and firefox. And we isolate creation & destroy of a context to one test suite, so I'm sure they never overlap.

**Comment by LanderBeeuwsaert** [comment]
@yury-s anyway, can I somehow easily output the trace to a file during the run? Now it's incomplete because it was outputting to the console & I copy/pasted it afterwards.

**Comment by yury-s** [comment]
> @yury-s anyway, can I somehow easily output the trace to a file during the run?

Yes, you can specify the file with `DEBUG_FILE=output.txt`. piping stderr output to a file should also work (something like `node script.js 2>stderrr.txt`)

**Comment by LanderBeeuwsaert** [comment]
@yury-s top! It worked. I have 2 runs that fail, however, both files are bigger than 10 MB so I can't send them through by github.
Is it possible to send them to you in another way?

**Comment by yury-s** [comment]
You can send to the email in my profile though I guess gmail has a limit of 25Mb per attachment so perhaps upload it to some file sharing service (google drive, one drive etc)?

**Comment by LanderBeeuwsaert** [comment]
@yury-s 
[output-for-webkit.zip](https://github.com/microsoft/playwright/files/5921738/output-for-webkit.zip)
[output-for-webkit-with-2-second-await-at-the-end.zip](https://github.com/microsoft/playwright/files/5921739/output-for-webkit-with-2-second-await-at-the-end.zip)
stupid me, zipping solves the problem

**Comment by mortelee** [comment]
Listening in,
Have similar problem with locked video file.
What I found out so far is that it looks that it has to do with the fact that I'm redirecting to a different page.
When using playwright-video I see that when the page is redirected the recording of the test immediately stops.

To see if you have similar problem you could try to run the test with playwright-video and see what happens.

**Comment by mortelee** [comment]
Do you guys have an update on this problem?

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

We are prioritizing the bugs based on the upvotes, recency and our ability to act. It looks like this issue only has a handful of upvotes, has not been touched recently and/or we lack sufficient feedback to act on it. We are closing issues like this one to keep our bug database maintainable. Please feel free to open a new issue and link this one to it if you think this is a mistake.
